lab_2/main.py

```diff
@@ -81,18 +208,113 @@ def calculate_text_plagiarism_score(original_text_tokens: tuple, suspicious_text
+                return -1.0
+    for sentence in suspicious_text_tokens:
+        if not isinstance(sentence, tuple):
+            return -1.0
+        for token in sentence:
+            if not isinstance(token, str) and not isinstance(token, int):
+                return -1.0
+
+    if len(original_text_tokens) < len(suspicious_text_tokens):
+        add = (len(suspicious_text_tokens) - len(original_text_tokens))
+        while add > 0:
+            original_text_tokens = list(original_text_tokens)
+            original_text_tokens.append(('',))
+            add -= 1
+        original_text_tokens = tuple(original_text_tokens)
+
+    total = 0.0
+    for sentence_index in range(len(suspicious_text_tokens)):
+        if len(suspicious_text_tokens) > 100:
+            lcs_length = find_lcs_length_optimized(original_text_tokens[sentence_index],
+                                                   suspicious_text_tokens[sentence_index], plagiarism_threshold)
+        else:
+            lcs_length = find_lcs_length(original_text_tokens[sentence_index], suspicious_text_tokens[sentence_index],
+                                         plagiarism_threshold)
+        total += calculate_plagiarism_score(lcs_length, suspicious_text_tokens[sentence_index])
+    result = total / len(suspicious_text_tokens)
+    if result < 0:
+        return 0.0
+    else:
+        return result
 
 
-def find_diff_in_sentence(original_sentence_tokens: tuple, suspicious_sentence_tokens: tuple, lcs: tuple) -> tuple:
+def find_diff_in_sentence(original_sentence_tokens: tuple, suspicious_sentence_tokens: tuple, lcs: tuple) -> tuple:  # +
     """
     Finds words not present in lcs.
     :param original_sentence_tokens: a tuple of tokens
     :param suspicious_sentence_tokens: a tuple of tokens
     :param lcs: a longest common subsequence
     :return: a tuple with tuples of indexes
     """
-    pass
+    checks = [
+        isinstance(original_sentence_tokens, tuple),
+        isinstance(suspicious_sentence_tokens, tuple),
+        isinstance(lcs, tuple)
+    ]
+    if not all(checks):
+        return ()
+    if lcs:
+        for token in lcs:
+            if not isinstance(token, str) and not isinstance(token, int):
+                return ()
+    if original_sentence_tokens:
+        for token in original_sentence_tokens:
+            if not isinstance(token, str) and not isinstance(token, int):
+                return ()
+    if suspicious_sentence_tokens:
+        for token in suspicious_sentence_tokens:
+            if not isinstance(token, str) and not isinstance(token, int):
+                return ()
+
+    result = []
+    orig_diff = []
+    for token_index, token in enumerate(original_sentence_tokens):
+        if token not in lcs:
+            orig_diff.append(token_index)
+            orig_diff.append(token_index + 1)
+    i_orig = 1
+    for element in orig_diff:
+        if element == orig_diff[-1]:
+            pass
```

Comment:
Do you need this `if` at all?

Reply:
yes, it is to avoid error "list index out of range"

Reply:
Then there should be another `key`-word.